### DISCUSSION TITLE
Following a Pattern

### DISCUSSION THREAD

**Discussion by railroadss** [PrismarineJS/mineflayer#1946] [General]
![image](https://user-images.githubusercontent.com/81101577/120079321-23854380-c068-11eb-9e4d-6c003718def7.png)
How could the bot track a pattern where it goes obsidian, air obsidian, air etc

**Comment by amoraschi** [comment]
What do you mean by "track"?

**Reply by railroadss** [reply]
Like, if block goes obsidian, air, obsidian. Basically, how do you get all of the boxes in that x or z

**Reply by amoraschi** [reply]
You could do a .findBlock and combine some if statements

**Reply by Karang** [reply]
Could you elaborate on your problem ? What do you want to do with this pattern ? Follow it ? Use it as a delimitation ? Is the pattern always the same ? How large it is ? is it a straight line or can it be an arbitrary path ? Are the blocks always 1 block appart ?

**Reply by railroadss** [reply]
Yeah, i just need to get the blocks in a line for x amount of distance the rest is pretty easy. I'm asking this as I am new to mineflayer. Is there a for loop for all of the blocks in a straight line?

**Reply by amoraschi** [reply]
No but you can do a for loop of blocks in a straight line, maybe with bot.blockAt
